webapp/components/post_view/post_message_view/post_message_view.jsx

```diff
@@ -176,7 +176,7 @@ export default class PostMessageView extends React.PureComponent {
             siteURL: this.props.siteUrl,
             mentionKeys: this.props.mentionKeys,
             atMentions: true,
-            channelNamesMap: getChannelsNameMapInCurrentTeam(store.getState()),
+            channelNamesMap: Object.assign({}, (this.props.post.props || {}).channel_mentions || {}, getChannelsNameMapInCurrentTeam(store.getState())),
```

Comment:
Is it possible to move this into a selector in the container so that it's not constantly remaking this object every time it renders? If not, can you at least split this onto a few lines so that it's clearer what it's doing like
```
let channelNamesMap = getChannelsNameMapInCurrentTeam(store.getState());
if (this.props.post.props && this.props.post.props.channl_mentions) {
    channelNamesMap = Object.assign({}, this.props.post.props.channel, channelNamesMap);
}
```

It would also be nice to move the call to getChannelsNameMapInCurrentTeam into the container as well, but that's outside the scope of your changes